### DISCUSSION TITLE
wireguard very slow speed

### DISCUSSION THREAD

**Discussion by longweihk** [angristan/wireguard-install#521] [Help]
wireguard have very slow speed , download only few mbit, upload 45mbit 
i try many MTU results not changed , have 1gbit connection download and upload

**Comment by adamk** [comment]
You will need to confirm the non-VPN download and upload speeds on both the client and server sides to know what should be expected. And are you running on top of any other VPNs? And what ISP(s) is being used? Any additional hardware being used or just a software Wireguard client?

**Comment by longweihk** [comment]
current server speedtest
Speedtest by Ookla

  Server: 4QH - Server Sponsoring - Dusseldorf (id: 61977)
     ISP: Contabo GmbH
Idle Latency: 0.97 ms (jitter: 0.04ms, low: 0.94ms, high: 1.02ms)
Download: 385.81 Mbps (data used: 183.6 MB)
66.32 ms (jitter: 10.92ms, low: 1.83ms, high: 128.72ms)
Upload: 373.53 Mbps [==\ ] 12% - latency: 102.50 ms Upload: 373.67 Mbps [==| ] 12% - latency: 102.50 ms Upload: 373.69 Mbps [==/ ] 13% - latency: 102.50 ms Upload: 373.81 Mbps [==- ] 14% - latency: 102.50 ms Upload: 373.64 Mbps [==\ ] 14% - latency: 102.50 ms Upload: 373.67 Mbps [===| ] 15% - latency: 102.50 ms Upload: 373.69 Mbps [===/ ] 16% - latency: 144.33 ms Upload: 373.71 Mbps [===- ] 16% - latency: 144.33 ms Upload: 373.97 Mbps [===\ ] 17% - latency: 144.33 ms Upload: 373.96 Mbps [===| ] 18% - latency: 144.33 ms Upload: 373.91 Mbps [===/ ] 18% - latency: 144.33 ms Upload: 374.16 Mbps [===- ] 19% - latency: 144.33 ms Upload: 374.05 Mbps [====\ ] 20% - latency: 172.40 ms Upload: 374.17 Mbps [====| ] 20% - latency: 172.40 ms Upload: 374.04 Mbps [====/ ] 21% - latency: 172.40 ms Upload: 374.20 Mbps [====- ] 22% - latency: 172.40 ms Upload: 374.01 Mbps [====\ ] 22% - latency: 172.40 ms Upload: 374.04 Mbps [====| ] 23% - latency: 172.40 ms Upload: 374.07 Mbps [=====/ ] 26% - latency: 116.20 ms Upload: 374.10 Mbps [======- ] 34% - latency: 116.20 ms Upload: 374.28 Mbps [========\ ] 41% - latency: 116.20 ms Upload: 374.21 Mbps [=========| ] 49% - latency: 116.20 ms Upload: 374.01 Mbps [===========/ ] 56% - latency: 116.20 ms Upload: 373.96 Mbps [============- ] 64% - latency: 116.20 ms Upload: 374.22 Mbps [==============\ ] 72% - latency: 116.20 ms Upload: 373.98 Mbps [===============| ] 79% - latency: 116.20 ms Upload: 374.10 Mbps [=================/ ] 87% - latency: 116.20 ms Upload: 374.28 Mbps [==================- ] 94% - latency: 116.20 ms Upload: 374.11 Mbps (data used: 168.3 MB)
116.20 ms (jitter: 35.94ms, low: 1.65ms, high: 315.82ms)
Packet Loss: 0.0%
Result URL: https://www.speedtest.net/result/c/6220343c-cff2-4168-a429-e90f369165bb

my current network speed test

i am using this wireguard client for windows 10 https://download.wireguard.com/windows-client/wireguard-installer.exe

https://www.speedtest.net/result/16820387480
Download Mbps
75.64
Upload Mbps
94.03
Ping ms 166 111 26
hardly test from speedtest great firewall, it gives this resuts,
speedtest.cn gives this results
![image](https://github.com/user-attachments/assets/f5f4e33b-c450-4e6b-8caa-440823c8bfa3)

**Reply by adamk** [reply]
Well now that makes a lot more sense. The GFW is very likely throttling your VPN traffic. You're lucky you're getting speeds that fast. You can always try installing Shadowsocks to obfuscate your Wireguard packets. That might help a little.

**Comment by longweihk** [comment]
no that not the issue, if that is the issue, not have very fast speed other vpn providers wireguard protocols, i also have astrill, i can connect without any problem wireguard protocols and speed is more than 200mbit, i try to solve this problem so i can use my server and not need any other vpn providers future
